src/content/docs/en/guides/upgrade-to/v3.mdx

```diff
@@ -0,0 +1,332 @@
+
+Update any existing references to `localhost:3000`, for example in tests or in your `README` to reflect the new port `localhost:4321`.
+
+### Changed: TypeScript minimum version
+
+In Astro v2.x, the `tsconfig.json` presets include support for both TypeScript 4.x and 5.x.
+
+Astro v3.0 updates the `tsconfig.json` presets to only support TypeScript 5.x. Astro now assumes that you use TypeScript 5.0 (March 2023), or that your editor includes it (e.g. VS Code 1.77)
+
+#### What should I do?
+
+Update your TypeScript version to at least v5.0.
+
+```bash
+npm install typescript@latest --save-dev
+```
+
+### Changed `astro check` is now an external package
+
+In Astro v2.x, `astro check` was included in Astro by default and its dependencies were bundled in Astro. This meant a larger package whether or not you ever used `astro check` and also prevented you from having control over the version of TypeScript and the Astro Language Server to use. 
+
+
+Astro v3.0 moves the `astro check` command out of Astro core and now requires an external package `@astrojs/check`. Additionally, you must install `typescript` in your project.
+
+#### What should I do?
+
+Run the `astro check` command after upgrading to Astro v3.0 and follow the prompts to install the required dependencies, or manually install `@astrojs/check` and `typescript` into your project.
+
+### Added: simple asset support for Cloudflare, Deno, Vercel Edge and Netlify Edge
+ In Astro v2.x, using the assets feature in Cloudflare, Deno, Vercel Edge and Netlify Edge errors in runtime as the environments do not support Astro's builtin Squoosh and Sharp image optimization.
+
+ Astro v3.0 allows these environments to work without errors, but does not perform any image transformation and processing. However, you would still get benefits, e.g. no Cumulative Layout Shift (CLS), enforced alt attribute, etc.
+
+ What should I do?
+ If you previously avoided the assets feature due these constraints, you can now use them without issues. You can also use the no-op image service to explicitly opt-in to this behaviour:
+
+ ```
+ // astro.config.mjs
+ export default {
+   image: {
+     service: {
+       entrypoint: 'astro/assets/services/noop'
+     }
+   }
+ }
+ ```
+
+In Astro v2.x, you would get an error log in development. When building a project, the build would fail.
+
+Astro V3 still displays the error log, but now you can build the project.
+
+When using an adapter that supports neither Squoosh or Sharp, Astro will now automatically use an image service that does not support processing, but still provides the other benefits of `astro:assets` such as enforcing `alt`, no CLS etc to users. Should be noted that it does no transformations or processing, but you still get all the other benefits (no CLS, alt enforced, etc). All of those benefits are part of the image service, if you make your own, you lose those benefits unless you make it yourself
+
+
+## Known Issues
+
+There are currently no known issues.
```

Comment:
``````suggestion
### Removed: `<Markdown />` component

In Astro v2.x, Astro deprecated the `<Markdown />` component and moved it to an external package.

Astro v3.0 comletely removes the package `@astrojs/markdown-component`, so Astro's `<Markdown />` component will no longer work in your project.

#### What should I do?

Uninstall the `@astrojs/markdown-component` package. To continue using a similar `<Markdown />` component in your code, consider using [community integrations](https://astro.build/integrations/) that provide a similar component like https://github.com/natemoo-re/astro-remote, and be sure to update your component imports and attributes as necessary according to the integration's own documentation. Otherwise, delete all references to importing Astro's `<Markdown />` component and the component itself in your `.astro` files. You will need to rewrite your content as HTML directly or [import Markdown](/en/guides/markdown-content/#importing-markdown) from an `.md` file.

### Removed: camelCase transformations

In Astro 2.x, Astro automatically transformed camelCase CSS variable names passed to the `style` attribute of an HTML element.

Astro 3.0 removes backwards-compatible kebab-case transform for these camelCase CSS variable names. 

#### What should I do?

If you were relying on Astro to transform kebab-case in your styles, update your existing styles to use the camelCase version to prevent missing styles. For example:

```astro
---
const myValue = "red"
---
<!-- input -->
<div style={{ "--myValue": myValue }}></div>
<!-- output (before) -->
<div style="--my-value:var(--myValue);--myValue:red"></div>
<!-- output (after) -->
<div style="--myValue:red"></div>
```

```diff
<style>
  div {
-   color: var(--my-value);
+   color: var(--myValue);
  }
</style>
```

## Known Issues

There are currently no known issues.

``````